internal/distributed/utils/util.go

```diff
@@ -30,3 +35,62 @@ func GracefulStopGRPCServer(s *grpc.Server) {
 		<-ch
 	}
 }
+
+func getTLSCreds(certFile string, keyFile string, nodeType string) credentials.TransportCredentials {
+	log.Info("TLS Server PEM Path", zap.String("path", certFile))
+	log.Info("TLS Server Key Path", zap.String("path", keyFile))
+	creds, err := credentials.NewServerTLSFromFile(certFile, keyFile)
+	if err != nil {
+		log.Warn(nodeType+" can't create creds", zap.Error(err))
+		log.Warn(nodeType+" can't create creds", zap.Error(err))
+	}
+	return creds
+}
+
+func EnableInternalTLS(NodeType string) grpc.ServerOption {
+	var Params *paramtable.ComponentParam = paramtable.Get()
+	var serverCfg *paramtable.GrpcServerConfig
+	switch strings.ToLower(NodeType) {
```

Comment:
Is this a good way?
I can add it as a parameter as well (NodeType string,ServerCfg *paramtable.GrpcServerConfig)